core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1243,6 +1248,43 @@ public boolean isNeededDependenciesCompatibleWithInstalledVersion(PluginManager.
             });
         }
 
+        /**
+         * Get the list of incompatible dependencies (if there are any, as determined by isNeededDependenciesCompatibleWithInstalledVersion)
+         *
+         * @since TODO
+         */
+        public List<Plugin> getDependenciesIncompatibleWithInstalledVersion() {
+            return getDependenciesIncompatibleWithInstalledVersion(new PluginManager.MetadataCache());
+        }
+
+        @Restricted(NoExternalUse.class) // table.jelly
+        @SuppressWarnings("unchecked")
+        public List<Plugin> getDependenciesIncompatibleWithInstalledVersion(PluginManager.MetadataCache cache) {
+            return cache.of("getDependenciesIncompatibleWithInstalledVersion:" + name, List.class, () -> {
```

Comment:
If we now cache the whole list of dependencies, we don't need to also cache the `bool` whether it's empty.

Reply:
Okay, I will rewrite the boolean version to just check whether the list is empty.